R/score.R

```diff
@@ -290,74 +135,44 @@ apply_metrics <- function(forecast, metrics, ...) {
 }
 
 
-#' Construct an object of class `scores`
+#' @title Validate metrics
+#'
 #' @description
-#' This function creates an object of class `scores` based on a
-#' data.table or similar.
-#' @param scores A data.table or similar with scores as produced by [score()].
-#' @param metrics A character vector with the names of the scores
-#'   (i.e. the names of the scoring rules used for scoring).
-#' @param ... Additional arguments to [data.table::as.data.table()]
-#' @keywords internal
-#' @importFrom data.table as.data.table setattr
-#' @return An object of class `scores`
-#' @examples
-#' \dontrun{
-#' df <- data.frame(
-#'   model = "A",
-#'   wis = "0.1"
-#' )
-#' new_scores(df, "wis")
-#' }
-new_scores <- function(scores, metrics, ...) {
-  scores <- as.data.table(scores, ...)
-  class(scores) <- c("scores", class(scores))
-  setattr(scores, "metrics", metrics)
-  return(scores[])
-}
-
-
-#' Create an object of class `scores` from data
-#' @description This convenience function wraps [new_scores()] and validates
-#'   the `scores` object.
-#' @inherit new_scores params return
-#' @importFrom checkmate assert_data_frame
-#' @keywords internal
-as_scores <- function(scores, metrics) {
-  assert_data_frame(scores)
-  present_metrics <- metrics[metrics %in% colnames(scores)]
-  scores <- new_scores(scores, present_metrics)
-  validate_scores(scores)
-  return(scores[])
-}
-
+#' This function validates whether the list of metrics is a list
+#' of valid functions.
+#'
+#' The function is used in [score()] to make sure that all metrics are valid
+#' functions.
+#'
+#' @param metrics A named list with metrics. Every element should be a scoring
+#'   function to be applied to the data.
+#' @importFrom cli cli_warn
+#'
+#' @return
+#' A named list of metrics, with those filtered out that are not
+#' valid functions
+#' @importFrom checkmate assert_list test_list check_function
+#' @keywords internal_input_check
+validate_metrics <- function(metrics) {
```

Comment:
Should this be in with the scores.default stuff?

Reply:
I was thinking that it is a helper function used by `score()` that's why I put it here. No strong feelings

Reply:
You mean put this into `class-scores.R`?

Reply:
Hm I really think it's just a helper for `score()`, so I don't think it makes much sense to put it in the `scores` class definition file

Reply:
So my assumption here is we would put things that are used within the method close to it but I think its also fine to keep as is